src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -5195,10 +5224,23 @@ void *wm_vuldet_main(wm_vuldet_t * vuldet) {
         }
 
         if ((curr_time - vuldet->last_scan >= vuldet->scan_interval)) {
+            if (!VU_TASKS_QUEUE_FLAGS[VU_INTERVAL_SCAN]) {
+                vu_task_ctx_t *task = create_task("%lu %d %d", 1, VU_INTERVAL_SCAN, WM_TASK_PENDING);
+                linked_queue_push_ex(vu_tasks_queue, task);
+            } else {
+                vuldet->last_scan = curr_time;
+            }
             wm_vuldet_run_scan(vuldet);
         }
 
-        wm_vuldet_run_sleep(vuldet);
+        /* Checks for messages in the queue */
+        vu_task_message_ctx_t *msg = (vu_task_message_ctx_t *)linked_queue_pop_ex_no_cond_wait(vu_messages_queue);
+        while(msg) {
+            /* TODO: Dummy message that represent interaction with task manager */
+            minfo("Task ID '%lu' status '%s' message '%s'", msg->task_id, task_statuses[msg->task_status], msg->status_message);
+            os_free(msg);
```

Comment:
We shouldn't forget in the future that the `vu_task_message_ctx_t` pointer also has allocated memory in `status_message`

Reply:
I added a check for that.

Reply:
What about if we create a helper function to free the messages as part of this PR? And we call it here instead of calling `os_free`.